File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
import math

def convert(n,k):
    arr=[]
    while n!=0:
        temp=n%k
        n//=k
        arr.append(temp)
    arr=arr[::-1]
    return "".join(list(map(str,arr)))

def check_prime(p):
    for i in range(2,int(math.sqrt(p))+1):
        if p%i==0:
            return False
    return True


def solution(n, k):
    answer = 0
    converted_n=convert(n,k)
    temp=converted_n.split("0")
    for s in temp:
        if s=="" or s=="1":
            continue
        num=int(s)
        if check_prime(num):
            answer+=1
    return answer

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;

class Solution {
    
    public int solution(int n, int k) {
        int answer = 0;
        
        String converted = Integer.toString(n, k);
        String[] arr = converted.split("0",-1);

        for(String tmp : arr){
            if(tmp.equals("")) continue;
            Long number = Long.valueOf(tmp);
            if(isPrime(number)){
                answer++;
            }
        }
        return answer;
    }
    
    private boolean isPrime(long number){
        if(number==1) return false;
        for(long i=2; i<=(long)Math.sqrt(number); i++){
            if(number%i==0) return false;
        }
        return true;
        
    }
}